docs/src/formula.md

```diff
@@ -279,6 +284,20 @@ Predictors:
   a(unknown) & b(unknown)
 ```
 
+!!! warning
+
+    Even though the `@formula` macro supports arbitrary julia functions,
+    runtime (programmatic) formula construction does not.  This is because to
+    resolve a symbol giving a function's _name_ into the actual _function_
+    itself, it's necessary to `eval`.  In practice this is not often an issue,
+    _except_ in cases where a package provides special syntax by overloading a
+    function (like `|` for
+    [MixedModels.jl](https://github.com/dmbates/MixedModels.jl), or `absorb`
+    for [Econometrics.jl](https://github.com/Nosferican/Econometrics.jl)).  In
+    these cases, you should use the corresponding constructors for the actual
+    terms themselves (e.g., `RanefTerm` and `FixedEffectsTerm` respectively), as
```

Comment:
How is the user supposed to discover these names?

It seems to me that a simpler alternative would be to ask package managers to directly overload `poly` `|` and `absorb` for terms, so that users can do `absorb(Terms(:a))`, `poly(Term(:a), 2)` etc. This also limits the namespace pollution (I guess I may also want to define a FixedEffectTerm in the FixedEffectModels package).

Reply:
Keep reading :)  I suppose it would be possible to include that in the main part of the example, too.

Reply:
(or rather, instead of having it separately down below)

Reply:
I've decided you're right and revised the whole section accordingly (including a note to make sure you export `poly`).  have another read?

Reply:
This looks great. Thanks!